src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -54,6 +58,58 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
+static int add_dissected_swap_cryptsetup(void) {
+
+#if HAVE_LIBCRYPTSETUP
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *d = NULL;
+        int r;
+
+        r = unit_name_from_path("/dev/disk/by-designator/swap-luks", ".device", &d);
```

Comment:
if the inputs are constant, you might as well hardcode the result here, it's easy enough to follow